model/src/plugin_handler.cpp

```diff
@@ -1180,22 +1199,32 @@ bool PluginHandler::ExtractMetadata(const std::string& path,
   std::string temp_path(tmpnam(0));
   if (!extractTarball(path, filelist, temp_path, true)) {
     std::ostringstream os;
-    os << "Cannot unpack plugin tarball at : " << path;
+    wxLogWarning("Cannot unpack plugin %s tarball at %s, check if tarball contains metadata.xml",
+      metadata.name.c_str(), path.c_str());
     if (filelist != "") cleanup(filelist, "unknown_name");
     last_error_msg = os.str();
     return false;
   }
-  if (!isRegularFile(temp_path.c_str())) return false;
+  if (!isRegularFile(temp_path.c_str())) {
+    // This could happen if the tarball does not contain the metadata.xml file
+    // or the metadata.xml file could not be extracted.
+    return false;
+  }
 
   struct CatalogCtx ctx;
   std::ifstream istream(temp_path);
   std::stringstream buff;
   buff << istream.rdbuf();
-  remove(temp_path.c_str());
-
+  int r = remove(temp_path.c_str());
+  if (r != 0) {
+    wxLogWarning("Cannot remove file %s: %s", temp_path.c_str(), strerror(r));
```

Comment:
Or, just to demonstrate: "MESSAGE_LOG << "Cannot remove file " << temp_path << ": " << strerror(r);"

The c++ style << actually has tons of advantages compared to the C-style %s format stuff.

Reply:
I changed all instances in this file to use `MESSAGE_LOG` or `DEBUG_LOG`.